includes/class.llms.frontend.forms.php

```diff
@@ -123,6 +130,58 @@ function mark_course_complete( $user_id, $lesson_id ) {
 			$value = 'yes';
 
 			$user_postmetas = $user->get_user_postmeta_data( $user_id, $course->id );
+			if ( ! empty( $user_postmetas['_is_complete'] ) ) {
+				if ( $user_postmetas['_is_complete']->meta_value === 'yes' ) {
+					return;
+				}
+			}
+
+			$update_user_postmeta = $wpdb->insert( $wpdb->prefix . 'lifterlms_user_postmeta',
+				array(
+					'user_id' 			=> $user_id,
+					'post_id' 			=> $course->id,
+					'meta_key'			=> $key,
+					'meta_value'		=> $value,
+					'updated_date'		=> current_time( 'mysql' ),
+				)
+			);
+
+			do_action( 'lifterlms_course_completed', $user_id, $course->id );
+
+		}
+	}
+
+	/**
+	 * mark section complete
+	 * Called by mark_lesson_complete
+	 *
+	 * If all lessons in section complete mark section as complete.
+	 *
+	 * @param  int $user_id [ID of the current user]
+	 * @param  int $lesson_id [ID of the current lesson]
+	 *
+	 * @return void
+	 * @version 3.2.4
+	 */
+	public function mark_section_complete( $user_id, $lesson_id ) {
```

Comment:
this can be removed in favor of student mark complete functions described earlier